### PR TITLE
feat(schematics): add the ability to create actions with the prefix

### LINKED ISSUE #3012: Allow to specify action name with custom prefix in actions schematics.
Currently, when generating action it's always prefixed with the "**load**", so that when I provide action name as "**DeleteUsers**", I will end-up with following action within generated file "**loadDeleteUsers**". It will be nice to have a possibility to generate action with the custom name without "**load**" prefix. In my case it would be "**deleteUsers**", "**deleteUsersSuccess**" etc. What's more I think it would be great to have an option to add actions to the existing file.

## If accepted, I would be willing to submit a PR for this feature

[X] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

### PR DESCRIPTION
**Pull request by mateuszbasinski**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

The actions are created only with the prefix Load.

Closes #3012 

## What is the new behavior?

The actions are created with a specified prefix.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: modules/schematics/src/action/schema.json [hunk 1]
```diff
@@ -13,6 +13,15 @@
       },
       "x-prompt": "What should be the name of the action?"
     },
+    "prefix": {
```
Comment (timdeschryver) [inline modules/schematics/src/action/schema.json:16]: Does this work? The `name` property already uses the unnamed argument, I think the default of this should be `load`, or the index should be 1 (but I prefer the default option)

File: modules/schematics/src/action/schema.ts [hunk 2]
```diff
@@ -4,6 +4,11 @@ export interface Schema {
    */
   name: string;
 
+  /**
+   * The name of the component.
```
Comment (timdeschryver) [inline modules/schematics/src/action/schema.ts:8]:
```suggestion
   * The prefix for the actions.
```

File: modules/schematics/src/action/schema.json (hunk 1 above)
Comment (mateuszbasinski) [reply modules/schematics/src/action/schema.json:16]: Yes, this works, but the default option is a better idea

**Comment by mateuszbasinski** [comment]
@timdeschryver I came up with the idea to generate actions with many prefixes by using one command. For example - for the action schematics we can pass an array with prefixes ([load, create, update, delete]) and then generate actions with all prefixes from the array. What do you think about this idea?

**Review by timdeschryver — APPROVED** [review]

**Comment by timdeschryver** [comment]
@mateuszbasinski I think that's a good idea.
I just don't know if the Angular CLI would parse a comma separated input correctly.
Have you tried that?

The only downside is that this leads to command-driven actions, while we encourage event-driven actions.
But the currently implementation also "violates" this practice 😅

**Comment by mateuszbasinski** [comment]
@timdeschryver I tried to deal with this using split. The output of splitted array looks good.
image

**Comment by mateuszbasinski** [comment]
@timdeschryver I tried to deal with this using split. The output of splitted array looks good.
image

**Comment by timdeschryver** [comment]
Then I think we can extend the schematic to create multiple defaults, as long as the default remains "load".

**Comment by brandonroberts** [comment]
I think this PR is good, but I don't think we should add comma-separated prefixes. As Tim mentioned, we're already violating event-driven actions, so I want to change those to be more correct, such as:

```
[Users Page] (Init/Load Users/Some other Event)
[Users Page/API] Users Loaded Success
[Users Page/API] Users Loaded Failure
```

I also think the goal of the schematics is to generate the minimum of what you need, and not try to auto-generate everything.

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**